src/plugin/contracts/plugins/SecurityPlugin.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.20;
+
+import '@cryptoalgebra/integral-core/contracts/libraries/Plugins.sol';
+
+import '@cryptoalgebra/integral-core/contracts/interfaces/IAlgebraPool.sol';
+
+import '../interfaces/IBasePluginV1Factory.sol';
+import '../interfaces/plugins/ISecurityPlugin.sol';
+import '../interfaces/plugins/ISecurityRegistry.sol';
+
+import '../base/BasePlugin.sol';
+
+/// @title Algebra Integral 1.2 security plugin
+abstract contract SecurityPlugin is BasePlugin, ISecurityPlugin {
+  using Plugins for uint8;
+
+  uint8 private constant defaultPluginConfig = uint8(Plugins.BEFORE_SWAP_FLAG | Plugins.BEFORE_FLASH_FLAG | Plugins.BEFORE_POSITION_MODIFY_FLAG);
+
+  address internal securityRegistry;
+
+  function _checkStatus(bool isBurn) internal {
+    ISecurityRegistry.Status status = ISecurityRegistry(securityRegistry).getPoolStatus(msg.sender);
+    if (status == ISecurityRegistry.Status.DISABLED) {
+      revert PoolDisabled();
+    } else if (status == ISecurityRegistry.Status.BURN_ONLY && isBurn == false) {
+      revert BurnOnly();
+    }
+  }
```

Comment:
For a small gas optimisation I think this function should be split into two: _checkStatus and _checkStatusOnBurn. Because currently it is more often called in beforeSwap with `isBurn = false` hardcoded and there is no need to check for BURN_ONLY status in that case